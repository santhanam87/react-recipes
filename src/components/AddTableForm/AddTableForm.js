import { useState } from "react";

export default function AddTableForm({ onFormSubmit }) {
  const [rows, setRows] = useState(0);
  const [columns, setColumns] = useState(0);
  const onSubmit = (event) => {
    event.preventDefault();
    onFormSubmit({ rows, columns });
  };
  return (
    <form className='form-container' onSubmit={onSubmit}>
      <div className='form-container'>
        <label>Rows</label>
        <input
          type='number'
          value={rows}
          onChange={(event) => {
            setRows(event.target.value);
          }}
        />
      </div>
      <div className='form-container'>
        <label>Columns</label>
        <input
          type='number'
          value={columns}
          onChange={(event) => {
            setColumns(event.target.value);
          }}
        />
      </div>
      <input type='submit' value='submitForm' />
    </form>
  );
}
