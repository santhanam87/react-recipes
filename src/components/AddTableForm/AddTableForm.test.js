import { render, fireEvent } from "@testing-library/react";
import AddTableForm from "./AddTableForm";

describe("AddTableForm", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(<AddTableForm onFormSubmit={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should call the mock onsubmit on formSubmit", () => {
    const mockSubmitCallback = jest.fn(() => {});
    const { getByText } = render(
      <AddTableForm onFormSubmit={mockSubmitCallback} />
    );
    const submitForm = getByText(/submitform/i);
    fireEvent.submit(submitForm);
    expect(mockSubmitCallback).lastCalledWith({ columns: 0, rows: 0 });
  });
});
