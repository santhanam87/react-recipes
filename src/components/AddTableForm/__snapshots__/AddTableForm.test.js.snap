// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTableForm should match the snapshot 1`] = `
<DocumentFragment>
  <form
    class="form-container"
  >
    <div
      class="form-container"
    >
      <label>
        Rows
      </label>
      <input
        type="number"
        value="0"
      />
    </div>
    <div
      class="form-container"
    >
      <label>
        Columns
      </label>
      <input
        type="number"
        value="0"
      />
    </div>
    <input
      type="submit"
      value="submitForm"
    />
  </form>
</DocumentFragment>
`;
